edgelet/hsm-sys/azure-iot-hsm-c/src/hsm_certificate_props.c

```diff
@@ -585,4 +612,99 @@ const char* get_alias(CERT_PROPS_HANDLE handle)
         result = handle->alias;
     }
     return result;
-}
\ No newline at end of file
+}
+
+int set_san_entries
+(
+    CERT_PROPS_HANDLE handle,
+    const char* san_list[],
+    size_t num_san_entries
+)
+{
+    int result;
+    if (handle == NULL || san_list == NULL || num_san_entries == 0)
+    {
+        LogError("Invalid parameter encounterered");
+        result = __LINE__;
+    }
+    else
+    {
+        size_t i;
+        size_t list_size = num_san_entries * sizeof(char*);
+        destroy_san_entries(handle);
+        if (((handle->san_list = (char **)malloc(list_size)) == NULL) ||
+            ((handle->san_list_ro = (char const**)malloc(list_size)) == NULL))
+        {
+            LogError("Could not allocate memory for SAN list");
+            result = __LINE__;
```

Comment:
What if the `san_list` allocation succeeds but the `san_list_ro` allocation fails? Will we leak `san_list`?

Reply:
Yes you are right, this is a possibility. I will fix this.

Reply:
Fixed and added negative test case as well to ensure coverage,